### DISCUSSION TITLE
Vulnerability fix for the kafka image

### DISCUSSION THREAD

**Discussion by Veljen** [strimzi/strimzi-kafka-operator#9704] [General]
hi,

we have found the below CVE for the kafka image.  can we fix this by updating the docker file? please guide me how to fix this ?

![image](https://github.com/strimzi/strimzi-kafka-operator/assets/120608238/fc37d323-8729-4a33-aeb7-349a53724edc)

**Comment by scholzj** [comment]
Strimzi builds on top of various other open-source projects and where possible, tries to update them every release. We do not rebuild all the dependencies from the source. If these vulnerabilities were updated in the source projects, they will likely be addressed in the next Strimzi release (which we expect shortly after Apache Kafka 3.7.0 is released). In the meantime, if you want, you can check the individual projects if they addressed these vulnerabilities and if not, contribute to fixes there. And if they did and the latest Strimzi builds from the main branch do not contain them yet, feel free to open a PR to update them.

**Reply by scholzj** [reply]
PS: You also don't even mention the image, so it is hard to see what exactly are you referring to.

**Reply by Veljen** [reply]
0.39.0 image only

**Reply by scholzj** [reply]
Every Strimzi release has several container images.

**Reply by Veljen** [reply]
quay.io/strimzi/kafka:0.39.0-0-kafka-3.6.1
quay.io/strimzi/operator:0.39.0

**Reply by Veljen** [reply]
@scholzj  can you please refer me the source projects and I seen the strimzi base image dockerfile and it has minimal package installation and some jar files were there. 

I am not sure which sourceproject I need to start with.

**Reply by scholzj** [reply]
The exact path where the vulnerability is found should normally tell you where it comes from. E.g. if it is in the Cruise Control folder, it is likely coming with Cruise Control etc.

**Reply by Veljen** [reply]
@scholzj 

https://github.com/apache/kafka/releases/tag/3.7.0   apache kafka released this version.  when strimzi will support this kafka version ?

**Reply by scholzj** [reply]
It is not yet released, it is not yet announced. The artifacts are just being staged for the release.

**Reply by Veljen** [reply]
@scholzj 

our security team is asking so much details.  who is the code signer for this repo?  I seen the maintainer page and its all from redhat and had personal email ID.. can you please help with this details?   redhat is officially code signer of this repo ?

**Reply by scholzj** [reply]
Strimzi is an open-source community. It is part of the Cloud Native Computing Foundation. Some people work on it for free in their spare time, others are paid by some companies and some mix both things together. But Strimzi does not belong to any particular company. Its maintainers can use whatever email they prefer.
